src/test/java/com/aws/greengrass/util/platforms/unix/UnixPlatformTest.java

```diff
@@ -5,28 +5,100 @@
 
 package com.aws.greengrass.util.platforms.unix;
 
+import com.aws.greengrass.logging.api.Logger;
+import com.aws.greengrass.logging.impl.LogManager;
 import com.aws.greengrass.testcommons.testutilities.GGExtension;
 import com.aws.greengrass.util.FileSystemPermission;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.condition.DisabledOnOs;
 import org.junit.jupiter.api.condition.OS;
 import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
 
+import java.io.IOException;
 import java.nio.file.attribute.PosixFilePermission;
+import java.util.Arrays;
 import java.util.Set;
 
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.arrayContaining;
 import static org.hamcrest.Matchers.contains;
 import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.containsString;
 import static org.hamcrest.Matchers.empty;
 import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertThrows;
 
-@ExtendWith({GGExtension.class})
+
+@ExtendWith({MockitoExtension.class, GGExtension.class})
 @DisabledOnOs(OS.WINDOWS)
 class UnixPlatformTest   {
 
     private static String[] command = {"echo", "hello", "world"};
+    private static final Logger logger = LogManager.getLogger(UnixPlatformTest.class);
+
+    @Spy
+    private final UnixPlatform spyPlatform = new UnixPlatform();
+    private final static String user = "ggc_user01";
+    private final static String group = "ggc_group01";
+
+    @Test
+    public void GIVEN_busybox_environment_WHEN_add_user_THEN_add_user_success() {
+        try {
+            spyPlatform.runCmd("useradd -r -m " + user, o -> {
+            }, "Failed to create user with useradd");
+        } catch (IOException e) {
+            logger.info(e.getMessage());
+        }
```

Comment:
What are you verifying with this? Seems like you're actually running these commands, so the test will behave differently depending on where it runs, and since you're catching and logging the exception, the test will never really fail but not accurately verify anything either, what purpose does this serve?

Reply:
Since the pull requests are to run on Ubuntu os (our automation test platform), but not the busybox env, then the way of thinking for the test case is as below 5 steps:
1. create a user named "ggc_user01" (the purpose is to let createUser method to run the busybox command "adduser")
2. call createUser method to create ggc_user01 (run "useradd" command), then IOException generated
3. Entered outter catch block, and run the busybox command "adduser"
4. Second IOException generated since busybox command "adduser" is running on Ubuntu os

![createUserMethod](https://user-images.githubusercontent.com/111258711/196331453-8dc28a21-3494-45d2-9433-cc6b94b91778.png)
![testcase](https://user-images.githubusercontent.com/111258711/196331458-ffb597a8-6a93-438a-8e99-f94effef5f88.png)